### DISCUSSION TITLE
How to keep "state" of map when temporarily hidden in view

### DISCUSSION THREAD

**Discussion by androbey** [alpinejs/alpine#1139] [1. Help] [answered]
Hi, 

I think it's more a general question if it is at all possible. So here's what I want to achieve: 

I am working with tabs using `<template x-if="tab == 'some_tab'"><!-- some HTML --></template>`.
In one "tab" I want to include an interactive map. However, if I interact with the map and choose a different tab and then come back to the tab with the map, the map gets re-drawn from scratch (so the "state of the map" (like position and zoom level) when I left the tab gets lost). Is there any way or workaround to prevent this behaviour? 

Hope some can help. 

Thanks.

**Comment by HugoDF** [comment, marked as answer]
Instead of `x-if` you can try to use `x-show` (you'll have to get rid of the `template` and apply x-show to the root element inside the template) which will toggle the visibility without removing the DOM nodes/elements

**Reply by androbey** [reply]
Seems kind of obvious now, but did not realize. Thanks, that worked!
